.github/workflows/build.yml

```diff
@@ -0,0 +1,80 @@
+name: build
+
+on: [push, pull_request, workflow_dispatch]
+# workflow_dispatch should make manually triggered ci/cd possible
+# workflow file (like this) with `workflow_dispatch` after on should exist on the **master** or default branch,
+# or there will be no ui for a manual trigger. https://github.community/t/workflow-dispatch-event-not-working/128856/2
+
+env:
+  GRADLE_OPTS: "-Xms128m -Xmx1024m"
+  ORG_GRADLE_PROJECT_logOutput: true
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
```

Comment:
Is this best practice to use the latest Ubuntu? Wouldn't it be better to specify a major version so that we can get stability and security bug fixes?

Reply:
According to some docs:
https://github.com/actions/virtual-environments#available-environments
https://github.blog/changelog/2020-10-29-github-actions-ubuntu-latest-workflows-will-use-ubuntu-20-04/,
it looks like in a foreseeable period of time, the `ubuntu-latest` will just be an alias of `ubuntu-20.04`, and it will use the LTS version ubuntu along with weekly updates for each version(e.g. https://github.com/actions/virtual-environments/commit/0da5da7f578a406c10b021472866a3e15c31f2ec)

I think github actions takes good care of the stability and security bug fixes, so from this point of view, it's safe to use either latest or any fixed version like `18.04`.

Reply:
At some point in the future, `ubuntu-latest` will change to a newer version of Ubuntu with unknown compatibility differences. I would prefer to the a fixed version `ubuntu-20.04` so that a surprise change like that does not break our build. We can choose later to update to a newer Ubuntu version when it becomes available, but we should keep this in our control. Also, Pravega uses the tag `ubuntu-20.04` in https://github.com/pravega/pravega/blob/master/.github/workflows/gradle.yml#L65.

Reply:
Changed to `ubuntu-20.04`.